### PR TITLE
Update OTLP auth example

### PR DESCRIPTION
**Pull request by damemi**
Add `x-goog-user-project` header and set `gcp.project_id` (not sure if that needs to be a span attribute or resource attribute, so I'm setting both for now)

Repro steps:
* `gcloud auth application-default login`
* `export OTEL_EXPORTER_OTLP_ENDPOINT=otlp://<endpoint>`
* `export GCLOUD_PROJECT=<project id>`
* `go run example.go`

### CODE REVIEW

**Review by aabmass — APPROVED** [review]

**Comment by damemi** [comment]
Updated this to add an http sample too, going to merge if everything is green

File: example/trace/otlpgrpc/example.go [hunk 1]
```diff
@@ -31,15 +34,18 @@ import (
...
 	creds, err := oauth.NewApplicationDefault(ctx)
 	if err != nil {
 		panic(err)
 	}
 
-	exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithDialOption(grpc.WithPerRPCCredentials(creds)))
+	res := resource.NewWithAttributes(semconv.SchemaURL, attribute.String("gcp.project_id", projectID))
+
+	// set endpoint with OTEL_EXPORTER_OTLP_ENDPOINT=https://<endpoint>
+	exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithDialOption(grpc.WithPerRPCCredentials(creds)), otlptracegrpc.WithHeaders(map[string]string{"x-goog-user-project": projectID}))
```
Comment (dashpole) [inline example/trace/otlpgrpc/example.go:48]: Should we tell users to use OTEL_EXPORTER_OTLP_HEADERS instead?

File: example/trace/otlpgrpc/example.go [hunk 2]
```diff
@@ -31,15 +34,18 @@ import (
...
+func initTracer(projectID string) (func(), error) {
 	ctx := context.Background()
 
 	creds, err := oauth.NewApplicationDefault(ctx)
 	if err != nil {
 		panic(err)
 	}
 
-	exporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithDialOption(grpc.WithPerRPCCredentials(creds)))
+	res := resource.NewWithAttributes(semconv.SchemaURL, attribute.String("gcp.project_id", projectID))
```
Comment (dashpole) [inline example/trace/otlpgrpc/example.go:45]: Can we use OTEL_RESOURCE_ATTRIBUTES instead?

File: example/trace/otlpgrpc/example.go (hunk 1 above)
Comment (damemi) [reply example/trace/otlpgrpc/example.go:48]: This took some digging, but it looks like `WithHeaders` overwrites OTEL_EXPORTER_OTLP_HEADERS (couldn't find docs but I found a [test asserting it](https://github.com/open-telemetry/opentelemetry-go/blob/e6e186bfa485f679e35bb775cba63ca24029590d/exporters/otlp/otlpmetric/otlpmetrichttp/internal/oconf/options_test.go#L333-L341), and I couldn't get it to work with both)

So if we pass Authorization using WithHeaders, we also need the other headers that way too. Unless they can be merged somehow

File: example/trace/otlpgrpc/example.go (hunk 2 above)
Comment (damemi) [reply example/trace/otlpgrpc/example.go]: yup, updated

**Pull request merged**